src/GCL/Serializer.cpp

```diff
@@ -24,21 +24,26 @@
 #include <iostream>
 
 #include <fcntl.h>
-#include <io.h>
 #include <string.h>
+#if defined(_WIN32)
+#include <io.h>
+#else
+#include <unistd.h>
+#include <stdio.h>
+#endif
 
 //////////////////////////////////////////////////////////////////////////
 
 GCL::Serializer::Serializer( const std::filesystem::path& rPath )
 {
+#if defined( _WIN32 )
 	constexpr int OPEN_FLAGS       = O_WRONLY | O_BINARY | O_TRUNC | O_CREAT;
 	constexpr int SHARE_FLAGS      = SH_DENYNO;
 	constexpr int PERMISSION_FLAGS = S_IREAD | S_IWRITE;
-
-#if defined( _WIN32 )
 	POSIX_CALL( _wsopen_s( &m_FileDescriptor, rPath.c_str(), OPEN_FLAGS, SHARE_FLAGS, PERMISSION_FLAGS ) );
 #else // _WIN32
-	POSIX_CALL( m_FileDescriptor = open( rPath.c_str(), ofstream, SHARE_FLAGS, PERMISSION_FLAGS ) );
+	constexpr int OPEN_FLAGS       = O_WRONLY | O_TRUNC | O_CREAT;
```

Comment:
why not O_BINARY?

Reply:
Also nonstandard - there is no distinction with normal `open` between binary and other data

Reply:
*on other systems

Reply:
Now windows uses O_BINARY but otherwise its excluded